### PR TITLE
London | May 2025 | Houssam Lahlah | Acoursework/sprint 2

### PR DESCRIPTION
**Pull request by HoussamLh**
**Check List**

- [x] I have committed my files one by one, on purpose, and for a reason  
- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title  
- [x] I have tested my changes  
- [x] My changes follow the style guide  
- [x] My changes meet the requirements of this task

**Changelist :** 

This PR contains solutions for the Sprint 2 coursework exercises:

**key-errors:** 
Implemented capitalise, convertToPercentage, and fixed square function errors.

**mandatory-debug:** 
Debugged and fixed multiply, sum, and getLastDigit functions.

**mandatory-implement:** 
Implemented calculateBMI, converted strings to upper case, and created reusable toPounds function.

**mandatory-interpret:** 
Added formatTimeDisplay function to convert seconds into HH:MM:SS format.

**stretch-extend:** 
Extended formatAs12HourClock function with edge case handling and added tests.

All functions have been tested with relevant inputs to ensure correctness.

**Questions**

Is my use of descriptive commit messages clear and helpful for reviewers?

Are there any improvements I could make in code style or function structure based on the style guide?

### CODE REVIEW

**Comment by cjyuan** [comment]
You made some changes to the "Sprint-1" folder in your `main` branch. As a result, when you created the `acoursework/sprint-2` branch from main, it inherited those changes. However, the updates in the "Sprint-1" folder are unrelated to the Sprint-2 exercise.

Can you revert the changes made in the "Sprint-1" folder?
One way to "revert the changes" in the "Sprint-1" folder is to replace the "Spint-1" folder in this branch by the "Sprint-1" folder from CYF's `main` branch: https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/tree/main

You can download CYF's `main` branch as a ZIP file and then copy/paste the "Sprint-1" folder from the ZIP file to your branch.

Note: Your `acoursework/sprint-3` branch is also affected in the similar way.

File: Sprint-2/3-mandatory-implement/3-to-pounds.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
-// In Sprint-1, there is a program written in interpret/to-pounds.js
+// Define a function that converts kilograms to pounds
+function toPounds(kg) {
+  const pounds = kg * 2.20462;
+  return pounds.toFixed(2); // rounds to 2 decimal places
```
Comment (cjyuan) [inline Sprint-2/3-mandatory-implement/3-to-pounds.js:4]: The pounds refer to the British currency.
This exercise the objective is to rewrite the code in `Sprint-1/3-mandatory-interpret/3-to-pounds.js` as a function.

File: Sprint-2/5-stretch-extend/format-time.js [hunk 2]
```diff
@@ -1,25 +1,36 @@
...
+    // midnight
+    return `12:${minutes} am`;
+  } else if (hours === 12) {
+    // noon
+    return `12:${minutes} pm`;
+  } else if (hours > 12) {
+    return `${hours - 12}:${minutes} pm`;
+  } else {
+    // 1am to 11am
+    return `${time} am`;
```
Comment (cjyuan) [inline Sprint-2/5-stretch-extend/format-time.js:15]: Currently `"08:00"` is converted to `"08:00 am"` but `"20:00"` is converted to `"8:00 pm"`.

Can you make the format of the return value more consistent?

**Review by cjyuan — COMMENTED** [review]
Code is good and explanation is clear.

Some of the code could use some improvement.

File: Sprint-2/3-mandatory-implement/3-to-pounds.js (hunk 1 above)
Comment (HoussamLh) [reply Sprint-2/3-mandatory-implement/3-to-pounds.js]: Thank you for the feedback. I understand now that the exercise is about converting pence to British pounds, not kilograms to pounds.

I rewrite the code in Sprint-1/3-mandatory-interpret/3-to-pounds.js as a function that takes a pence string (like "399p") and returns the formatted pounds (like "£3.99"), then test it with different inputs.

Let me know if I’ve understood it correctly.

File: Sprint-2/5-stretch-extend/format-time.js (hunk 2 above)
Comment (HoussamLh) [reply Sprint-2/5-stretch-extend/format-time.js]: Thank you for the feedback! I understand now that the formatting should be consistent, so both the hour and minute parts should always have two digits (e.g., "08:00 am" instead of "8:00 am"). I've been updated my function to pad the hour with a leading zero when needed to make the output format uniform.

File: Sprint-2/5-stretch-extend/format-time.js [hunk 3]
```diff
@@ -1,25 +1,44 @@
...
+  } else if (hours > 12) {
+    formattedHour = String(hours - 12).padStart(2, "0");
+    period = "pm";
+  } else {
+    formattedHour = String(hours).padStart(2, "0");
+    period = "am";
   }
-  return `${time} am`;
+
+  return `${formattedHour}:${minutes} ${period}`;
```
Comment (cjyuan) [inline Sprint-2/5-stretch-extend/format-time.js:22]: Could also consider format the hours with `padStart(2, "0")` at line 22 (at one location); less code and make code maintenance easier.

**Review by cjyuan — COMMENTED** [review]
Changes look good and good job in fixing the branch.

- Your forgot to check these checkboxes
image

- You haven't yet changed the label to "Needs review" but I assumed you have finished changing your code.
